scripts/BuildAndTest.ps1

```diff
@@ -212,11 +212,18 @@ if (-not $?) {
 
 if (-not $NoBuild) {
     Invoke-DotNetBuild $SolutionFile
-    Invoke-DotNetBuild $sampleSolutionFile
+    if ($ENV:OS) {
+        Invoke-DotNetBuild $sampleSolutionFile
+    }
 }
 
 if (-not $NoTest) {
-    & dotnet test $SourceRoot\$SolutionFile --no-build --configuration $Configuration
+    if ($ENV:OS) {
+        & dotnet test $SourceRoot\$SolutionFile --no-build --configuration $Configuration
+    }
+    else {
+        & dotnet test $SourceRoot\$SolutionFile --no-build --configuration $Configuration --filter WindowsOnly!=true
```

Comment:
>--filter WindowsOnly!=true [](http://example.com/codeflow?start=90&length=26)

Consider defining this conditions as $LinuxOptions or something with a default value of null/empty string.

Reply:
added using $NonWindowsOptions